model/job.go

```diff
@@ -295,21 +307,24 @@ func (j *Job) WriteConfigs(role *Role, lightOpinionsPath, darkOpinionsPath strin
 		Networks   struct {
 			Default map[string]string `json:"default"`
 		} `json:"networks"`
-		ExportedProperties []string                    `json:"exported_properties"`
-		Consumes           map[string]*JobLinkConsumes `json:"consumes"`
+		ExportedProperties []string               `json:"exported_properties"`
+		Consumes           map[string]jobLinkInfo `json:"consumes"`
 	}
 
 	config.Job.Templates = make([]jobConfigTemplate, 0)
 	config.Parameters = make(map[string]string)
 	config.Properties = make(map[string]interface{})
 	config.Networks.Default = make(map[string]string)
 	config.ExportedProperties = make([]string, 0)
-	config.Consumes = make(map[string]*JobLinkConsumes, 0)
+	config.Consumes = make(map[string]jobLinkInfo)
 
 	config.Job.Name = role.Name
 
-	for _, roleJob := range role.Jobs {
+	for _, roleJob := range role.RoleJobs {
 		config.Job.Templates = append(config.Job.Templates, jobConfigTemplate{roleJob.Name})
+		for _, consumer := range roleJob.ResolvedConsumers {
```

Comment:
Why are the consumers for *all* jobs of this role added here? It looks like the output should only describe this particular job (`j`) and not all of them.

Reply:
Oops, see previous reply. We have a `Job j`, which doesn't have resolved consumers (that's on `RoleJob`). We could filter to only the `RoleJob`s that have a `Job` of `j`?

Reply:
Why not turn this into a method on a `RoleJob`? The calling site already calls it that way:

https://github.com/SUSE/fissile/blob/marky-links-multiple-providers/builder/role_image.go#L163

It just turns into a `Job` method because the `RoleJob` has an unnamed `*Job` member that obviously has already been filled in.